cms/static/cms/js/modules/cms.sideframe.js

```diff
@@ -196,58 +228,96 @@
 
                 // otherwise do normal behaviour
                 if (animate) {
-                    this.sideframe.animate({
+                    this.ui.sideframe.animate({
                         width: width,
                         overflow: 'visible'
                     }, this.options.sideframeDuration);
                 } else {
-                    this.sideframe.css('width', width);
+                    this.ui.sideframe.css('width', width);
                     // reset width if larger than available space
                     if (width >= $(window).width()) {
-                        this.sideframe.animate({
+                        this.ui.sideframe.css({
                             width: $(window).width() - 30,
                             overflow: 'visible'
-                        }, 0);
+                        });
                     }
                 }
 
-                // lock toolbar, set timeout to make sure CMS.API is ready
-                setTimeout(function () {
+                // trigger API handlers
+                if (CMS.API && CMS.API.Toolbar) {
+                    // FIXME: initialization needs to be done after our libs are loaded
```

Comment:
yep, the goal is to eliminate these race conditions using proper initialisation. The UI is not affected currently like that.